docs/auth.md

```diff
@@ -0,0 +1,66 @@
+---
+id: auth
+title: "Authorization with OAuth2"
+sidebar_label: "OAuth2"
+---
+
+**ZIO CLI** can perform interactions requiring OAuth2. OAuth2 is added to a CLI App as an `Options[OAuth2Token]`. The token can be stored and the user can specify the path, so it is not necessary to repeat authentication. We can create it as with other options:
```

Comment:
It's good to provide some motivating background information here: if the user is building a command-line application that needs to be authenticated (example: `git` command requires authentication), then one possible way to do that is OAuth 2. Provide a little background on OAuth 2 and then go on to discuss the feature and how to use it.